File: src/main/java/com/saju/manse_api/ManseController.java
package com.saju.manse_api;

import com.saju.manse_api.service.SajuResult;
import com.saju.manse_api.service.SajuService;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@CrossOrigin(origins = {"https://easysajusaju-dev.github.io"}) // GitHub Pages에서 호출 허용
public class ManseController {

text

@GetMapping("/ping")
public String ping() { return "pong"; }

@GetMapping("/saju")
public SajuResult saju(
        @RequestParam int year,
        @RequestParam int month,
        @RequestParam int day,
        @RequestParam int hour,
        @RequestParam(name = "min", defaultValue = "0") int min,
        @RequestParam(name = "isLunar", defaultValue = "false") boolean isLunar,
        @RequestParam(name = "leap", defaultValue = "false") boolean leap,
        @RequestParam(name = "isMale", defaultValue = "true") boolean isMale,
        @RequestParam(name = "pivotMin", defaultValue = "30") int pivotMin
) {
    return SajuService.INSTANCE.getSaju(year, month, day, hour, min, isLunar, leap, isMale, pivotMin);
}
}

File: src/main/java/com/saju/manse_api/ManseController.kt
package com.saju.manse_api

import com.saju.engine.SajuFullEngine
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/saju")
class ManseController {

    // ------------------------------------------------------
    // 🔥 통합 사주 API — 프론트는 이것만 호출하면 됨
    // ------------------------------------------------------
    @GetMapping("/full")
    fun fullSaju(
        @RequestParam year: Int,
        @RequestParam month: Int,
        @RequestParam day: Int,
        @RequestParam hour: Int,
        @RequestParam min: Int,
        @RequestParam isLunar: Boolean,
        @RequestParam leap: Boolean,
        @RequestParam isMale: Boolean
    ): Any {
        return try {
            val res = SajuFullEngine.run(
                year = year,
                month = month,
                day = day,
                hour = hour,
                minute = min,
                isLunar = isLunar,
                leap = leap,
                isMale = isMale
            )
            mapOf("ok" to true, "result" to res)
        } catch (e: Exception) {
            mapOf("ok" to false, "error" to e.message)
        }
    }


    // ------------------------------------------------------
    // 🔥 Ping — 서버 깨우기용
    // ------------------------------------------------------
    @GetMapping("/ping")
    fun ping() = "pong"
}
